analysis/variables_cohorts.py

```diff
@@ -0,0 +1,584 @@
+            )
+        .date
+        ),
+
+      ### Asthma
+        out_date_asthma= (
+            first_matching_event_clinical_snomed_between(
+            asthma_snomed, index_date, end_date_out
+            )
+        .date
+        ),
+
+      ### COPD
+        out_date_copd= (
+            first_matching_event_clinical_ctv3_between(
+            copd_ctv3, index_date, end_date_out
+            )
+        .date
+        ),
+
+      ### Pulmonary Fibrosis
+        out_date_pulmonary_fibrosis= (
+            first_matching_event_clinical_snomed_between(
+            pulmonary_fibrosis_snomed, index_date, end_date_out
+            )
+        .date
+        ),
+
+# DEFINE EXISTING RESPIRATORY CONDITION COHORT --------------------------------------------------------------
+        sub_bin_asthma_recent_snomed= (
+            first_matching_event_clinical_ctv3_between(
+            copd_ctv3, index_date-days(730), index_date-days(1)
+            )
+        .exists_for_patient()        
+        ),
+# COPD diagnosed ever----------------------------------------------------------------------------------------       
+        sub_bin_copd_ctv3=(
+            last_matching_event_clinical_ctv3_before(
+            copd_ctv3, index_date
+            )    
+        .exists_for_patient()       
+        ),
+
+# Covariates-------------------------------------------------------------------------------------------------  
+
+        ## Age
+        cov_date_of_birth=patients.date_of_birth,
+        cov_num_age=patients.age_on(index_date),
+
+        ## Sex
+        cov_cat_sex=patients.sex,
+
+        ## Ethnicity
+        cov_cat_ethnicity=(
+            clinical_events.where(
+                clinical_events.ctv3_code.is_in(opensafely_ethnicity_codes_6)
+            )
+            .sort_by(clinical_events.date)
+            .last_for_patient()
+            .ctv3_code.to_category(opensafely_ethnicity_codes_6)
+        ),
+
+        ## Deprivation (IMD, 5 categories)
+        cov_cat_imd=case(
+            when((addresses.for_patient_on(index_date).imd_rounded >= 0) & 
+                 (addresses.for_patient_on(index_date).imd_rounded < int(32844 * 1 / 5))).then("1 (most deprived)"),
+            when(addresses.for_patient_on(index_date).imd_rounded < int(32844 * 2 / 5)).then("2"),
+            when(addresses.for_patient_on(index_date).imd_rounded < int(32844 * 3 / 5)).then("3"),
+            when(addresses.for_patient_on(index_date).imd_rounded < int(32844 * 4 / 5)).then("4"),
+            when(addresses.for_patient_on(index_date).imd_rounded < int(32844 * 5 / 5)).then("5 (least deprived)"),
+            otherwise="unknown",
+        ),
+
+        ## Region
+        cov_cat_region=practice_registrations.for_patient_on(index_date).practice_nuts1_region_name,
+        ## Consultation rate (these codes can run locally but fail in GitHub action test, details see https://docs.opensafely.org/ehrql/reference/schemas/tpp/#appointments)
```

Comment:
It is okay that they fail the test - OpenSAFELY are aware of why we are using this variable and are happy for us to continue using it.